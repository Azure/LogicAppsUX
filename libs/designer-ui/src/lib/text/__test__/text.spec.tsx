import * as React from 'react';
import { describe, beforeEach, it, expect } from 'vitest';
import { LargeText, MediumText, SmallText, TextProps, XLargeText, XXLargeText } from '../index';
import renderer from 'react-test-renderer';

describe('libs/designer-ui/text', () => {
  let minimal: TextProps;

  beforeEach(() => {
    minimal = {
      text: 'Hello Logic Apps',
      className: 'text-classname-test',
    };
  });

  it('Should render a MediumText component without styles and classnames', () => {
    const textComponent = <MediumText text={'Hello Logic Apps'} />;
    const renderedComponent = renderer.create(textComponent).toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it('Should render a MediumText component with a specific style display block and a border of 1px red solid', () => {
    const textComponent = <MediumText {...minimal} style={{ display: 'block', border: '1px red solid' }} />;
    const renderedComponent = renderer.create(textComponent).toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it('Should render a XXLargeText component', () => {
    const textComponent = <XXLargeText {...minimal} />;
    const renderedComponent = renderer.create(textComponent).toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it('Should render a XLargeText component', () => {
    const textComponent = <XLargeText {...minimal} />;
    const renderedComponent = renderer.create(textComponent).toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it('Should render a LargeText component', () => {
    const textComponent = <LargeText {...minimal} />;
    const renderedComponent = renderer.create(textComponent).toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it('Should render a MediumText component', () => {
    const textComponent = <MediumText {...minimal} />;
    const renderedComponent = renderer.create(textComponent).toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it('Should render a SmallText component', () => {
    const textComponent = <SmallText {...minimal} />;
    const renderedComponent = renderer.create(textComponent).toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });
});
