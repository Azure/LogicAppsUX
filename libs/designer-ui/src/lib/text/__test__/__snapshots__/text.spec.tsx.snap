// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`libs/designer-ui/text > Should render a LargeText component 1`] = `
<span
  className="fui-Text text-classname-test ___1kd1avm_ql4jt80 fk6fouc fod5ikn faaz57k figsok6 fpgzoln f1w7gpdv f6juhto f1mtd64y f1y7q3j9 f2jf649"
>
  Hello Logic Apps
</span>
`;

exports[`libs/designer-ui/text > Should render a MediumText component 1`] = `
<span
  className="fui-Text text-classname-test ___nzdt000_bc09y70 fk6fouc fkhj508 f1i3iumi figsok6 fpgzoln f1w7gpdv f6juhto f1mtd64y f1y7q3j9 f2jf649"
>
  Hello Logic Apps
</span>
`;

exports[`libs/designer-ui/text > Should render a MediumText component with a specific style display block and a border of 1px red solid 1`] = `
<span
  className="fui-Text text-classname-test ___nzdt000_bc09y70 fk6fouc fkhj508 f1i3iumi figsok6 fpgzoln f1w7gpdv f6juhto f1mtd64y f1y7q3j9 f2jf649"
  style={
    {
      "border": "1px red solid",
      "display": "block",
    }
  }
>
  Hello Logic Apps
</span>
`;

exports[`libs/designer-ui/text > Should render a MediumText component without styles and classnames 1`] = `
<span
  className="fui-Text ___nzdt000_bc09y70 fk6fouc fkhj508 f1i3iumi figsok6 fpgzoln f1w7gpdv f6juhto f1mtd64y f1y7q3j9 f2jf649"
>
  Hello Logic Apps
</span>
`;

exports[`libs/designer-ui/text > Should render a SmallText component 1`] = `
<span
  className="fui-Text text-classname-test ___grx76c0_ewuzpv0 fk6fouc fy9rknc fwrc4pm figsok6 fpgzoln f1w7gpdv f6juhto f1mtd64y f1y7q3j9 f2jf649"
>
  Hello Logic Apps
</span>
`;

exports[`libs/designer-ui/text > Should render a XLargeText component 1`] = `
<span
  className="fui-Text text-classname-test ___1l1yvat_1ip91ff fk6fouc f1pp30po f106mvju fdj6btp fpgzoln f1w7gpdv f6juhto f1mtd64y f1y7q3j9 f2jf649"
>
  Hello Logic Apps
</span>
`;

exports[`libs/designer-ui/text > Should render a XXLargeText component 1`] = `
<span
  className="fui-Text text-classname-test ___17o1977_19zs942 fk6fouc fojgt09 fcen8rp fl43uef fpgzoln f1w7gpdv f6juhto f1mtd64y f1y7q3j9 f2jf649"
>
  Hello Logic Apps
</span>
`;
